### ISSUE TITLE
Segment in a clipping box

### ISSUE THREAD

**Issue by ORNis** [cloudcompare/trunk#215]
Hello Daniel,
I'm making a lot of scan cleanning with CloudCompare at this time. My typical workflow consists of segmenting my point cloud until I'm sure that when I will create the polygon around the part I want to trash, the extrusion of the polygon wont "hit" some good part of the scan. So, I was thinking it would be great to have a kind of temporary clipping box in the manual segmentation toolbox just to bound the extrusion of the polygon.
Also, it could be nice if the actual clipping box mechanism could also segment a point cloud and not only making a copy of the selected part.
What do you think of these ideas ?
Thx,
Romain

**Comment by cloudcompare** [comment]
Hi Romain!

For the 'temporary clipping box', how do you imagine it? Just as the box of the 'Clipping-box' tool? Or maybe something simpler?

Otherwise the second idea is rather easy. It shouldn't be too hard to implement.

Daniel

**Comment by ORNis** [comment]
Yes maybe it could use  the same box as the "clipping-box" tool. The user could have the choice to  to toggle interactors in the segmentation toolbox and also to reset the clipping box (and maybe clip inside/outside ?)... it seems "easy" to add but It could be messy if the user have selected multiple clouds.

**Comment by ryukisai** [comment]
Hi all, @ OP, this issue will be partially resolved by the newly implemented (https://github.com/CloudCompare/CloudCompare/issues/1686).
However, your suggestion to use the clipping box to segment a cloud and not only copy the selected points is great and still is valid today.
I don't know if my comment is useful here 7 years later but here i am…
